### PR TITLE
Remove --dry-run from Kani.

### LINKED ISSUE #1936: Replace `--dry-run` by `--print-commands`
**Proposed change:** Replace the `--dry-run` option by something like `--print-commands`. Or even just `--verbose` (like `cargo`).

**Motivation:**

1.  First, this would be much simpler and we would be able to remove a bunch of hacks we add in `kani-driver` to support `--dry-run`. 
2. Usually when I run `--dry-run` is because I want to reproduce a similar sequence of commands with Kani, since `--dry-run` uses mock data, it forces me to edit the lines I get from Kani.
3. I also end up having to run Kani twice. `kani --dry-run` followed by a `kani --keep-temps`. This would make it easier to do one pass of commands only.

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

I also ensured that `--verbose` was always printing the commands `kani-driver` invokes and I added a `Running:` prefix to the command being printed so they are grepable.

### Resolved issues:

Resolves #1936

### Call-outs:

I kept the `--dry-run` and turned into an error for now as a reminder to people.

### Testing:

* How is this change tested? Modified tests

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-driver/src/args.rs [hunk 1]
```diff
@@ -448,6 +446,13 @@ impl KaniArgs {
             ));
         }
 
+        if self.dry_run {
+            return Err(Error::raw(
+                ErrorKind::ValueValidation,
+                "We no longer support `--dry-run`. Use --verbose instead.",
```
Comment (zhassan-aws) [inline kani-driver/src/args.rs:452]: Suggest saying:
```
The `--dry-run` option has been deprecated. Use `--verbose` instead.
```

**Review by zhassan-aws — APPROVED** [review]
Looks good. Thanks!

It might be useful to add a Kani-specific prefix to the "Running:" line, e.g.
```
[Kani] Running: ...
```
but the current form is fine too.

Also, might want to say "Executing command:" intead of "Running".

File: kani-driver/src/args.rs (hunk 1 above)
Comment (celinval) [reply kani-driver/src/args.rs:452]: Usually deprecated  means that it still works and it will be removed which isn't the case here.

**Comment by celinval** [comment]
> Looks good. Thanks!
> 
> It might be useful to add a Kani-specific prefix to the "Running:" line, e.g.
> 
> ```
> [Kani] Running: ...
> ```
> 
> but the current form is fine too.
> 
> Also, might want to say "Executing command:" intead of "Running".

Sure, I can add a tag. I do prefer `Running` though since it is shorter and it matches cargo output. Is that OK?

File: kani-driver/src/args.rs (hunk 1 above)
Comment (zhassan-aws) [reply kani-driver/src/args.rs:452]: How about "obsolete"?

**Comment by zhassan-aws** [comment]
> Sure, I can add a tag. I do prefer `Running` though since it is shorter and it matches cargo output. Is that OK?

Sure.

File: kani-driver/src/args.rs (hunk 1 above)
Comment (celinval) [reply kani-driver/src/args.rs:452]: Sure

**Pull request merged**